File: bookOfProblems/leetcode.com/golang/two-pointers/2000. Reverse Prefix of Word.go
func reversePrefix(word string, ch byte) string {
	runes := []rune(word)

	for l, r := 0, 0; r < len(runes); r++ {
		if runes[r] == rune(ch) {
			for l <= r {
				runes[l], runes[r] = runes[r], runes[l]
				l++
				r--
			}
			break
		}
	}
	return string(runes)
}

////////////////////////////////////////////////////////////////////////

func reversePrefix(word string, ch byte) string {
	prefix := ""
	for i := 0; i < len(word); i++ {
		prefix = string(word[i]) + prefix
		if word[i] == ch {
			prefix += word[i+1:]
			return prefix
		}
	}

	return word
}

File: bookOfProblems/leetcode.com/javascript/2000. Reverse Prefix of Word.js
word = "xyxzxe", ch = "z"
/**
 * @param {string} word
 * @param {character} ch
 * @return {string}
 */
var reversePrefix = function (word, ch) {

    let chIndex = word.indexOf(ch)
    let prefix = word.slice(0, chIndex + 1).split('').reverse().join('')
    let postfix = word.slice(chIndex + 1)

    return prefix.concat(postfix)
};

console.log(reversePrefix(word, ch));

////////////////////////////////////////////////////////////////////////////////

var reversePrefix = function (word, ch) {
    const index = word.indexOf(ch);
    return word.substr(0, index + 1).split('').reverse().join('') + word.substr(index + 1);
};